bindings/python/iota_sdk/types/block/wrapper.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2023 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Dict, Union
+from dacite import from_dict
+from iota_sdk.types.block.basic import BasicBlock
+from iota_sdk.types.block.validation import ValidationBlock
+from iota_sdk.types.common import HexStr, json
+from iota_sdk.types.signature import Ed25519Signature
+from iota_sdk.utils import Utils
+
+
+@json
+@dataclass
+class BlockWrapper:
+    """A block wrapper type that can hold either a `BasicBlock` or a `ValidationBlock`.
+    Shared data is stored alongside such a block in the `BlockHeader` and `Signature` fields.
+
+    Attributes:
+        header: The Block header.
+        block: Holds either a `BasicBlock` or a `ValidationBlock`.
+        signature: The Block signature.
+    """
+    protocol_version: int
+    network_id: str
+    issuing_time: str
+    slot_commitment_id: HexStr
+    latest_finalized_slot: int
+    issuer_id: HexStr
+    block: Union[BasicBlock, ValidationBlock]
+    signature: Union[Ed25519Signature]
```

Comment:
Any reason this is a union type?

Reply:
you mean the `signature`, right? Because there will potentially be various signatures, not just Ed25519?

Reply:
are we sure there ever will be?

Reply:
the design seems to be flexible in that regard:
```python
@json
@dataclass
class Signature():
    """Base class of a signature.
    """
    type: int


@json
@dataclass
class Ed25519Signature(Signature):
    """An Ed25519 signature.

    Attributes:
        public_key: The Ed25519 public key.
        signature: The Ed25519 signature of some message.
        type: The Ed25519 signature type.
    """
    public_key: HexStr
    signature: HexStr
    type: int = field(default_factory=lambda: 0, init=False)
```

Reply:
Yeah, but in python the type wouldn't be breaking to make it a union later afaik

Reply:
removed the union

Reply:
added the Union back :trollface: